awswrangler/opensearch/_utils.py

```diff
@@ -37,13 +51,87 @@ def _is_https(port: int) -> bool:
     return port == 443
 
 
+def _get_default_encryption_policy(collection_name: str, kms_key_arn: Optional[str]) -> Dict[str, Any]:
+    policy: Dict[str, Any] = {
+        "Rules": [
+            {
+                "ResourceType": "collection",
+                "Resource": [
+                    f"collection/{collection_name}",
+                ],
+            }
+        ],
+    }
+    if kms_key_arn:
+        policy["KmsARN"] = kms_key_arn
+    else:
+        policy["AWSOwnedKey"] = True
+    return policy
+
+
+def _get_default_network_policy(collection_name: str, vpc_endpoints: Optional[List[str]]) -> List[Dict[str, Any]]:
+    policy: List[Dict[str, Any]] = [
+        {
+            "Rules": [
+                {
+                    "ResourceType": "dashboard",
+                    "Resource": [
+                        f"collection/{collection_name}",
+                    ],
+                },
+                {
+                    "ResourceType": "collection",
+                    "Resource": [
+                        f"collection/{collection_name}",
+                    ],
+                },
+            ],
+            "Description": f"Default network policy for collection '{collection_name}'.",
+        }
+    ]
+    if vpc_endpoints:
+        policy[0]["SourceVPCEs"] = vpc_endpoints
+    else:
+        policy[0]["AllowFromPublic"] = True
+    return policy
+
+
+def _create_security_policy(
+    collection_name: str,
+    policy: Optional[Union[Dict[str, Any], List[Dict[str, Any]]]],
+    type: str,  # pylint: disable=redefined-builtin
+    client: boto3.client,
+    **kwargs: Any,
+) -> None:
+    if not kwargs:
+        kwargs = {}
+    if type == "encryption" and not policy:
+        policy = _get_default_encryption_policy(collection_name, kwargs.get("kms_key_arn"))
+    elif type == "network" and not policy:
+        policy = _get_default_network_policy(collection_name, kwargs.get("vpc_endpoints"))
+    else:
+        raise exceptions.InvalidArgument(f"Invalid policy type '{type}'.")
+    try:
+        client.create_security_policy(
+            name=f"{collection_name}-{type}-policy",
+            policy=json.dumps(policy),
+            type=type,
+            description=f"Default {type} policy for collection '{collection_name}'.",
+        )
+    except botocore.exceptions.ClientError as error:
+        if error.response["Error"]["Code"] == "ConflictException":
+            raise exceptions.Conflict("The policy name or rules conflict with an existing policy.") from error
+        raise error
+
+
 def connect(
     host: str,
     port: Optional[int] = 443,
     boto3_session: Optional[boto3.Session] = None,
     region: Optional[str] = None,
     username: Optional[str] = None,
     password: Optional[str] = None,
+    service: Optional[str] = None,
```

Comment:
What is the point of this argument if we should always be able to infer it from the host?

Reply:
We won't. AWS-managed OpenSearch would conform to the convention but we can't guarantee that if it's customer-managed instance, or if it's behind a proxy, for example.